packages/nuxt/src/core/plugins/layer-aliasing.ts

```diff
@@ -0,0 +1,69 @@
+import { createUnplugin } from 'unplugin'
+import type { NuxtConfigLayer } from 'nuxt/schema'
+import { resolveAlias } from '@nuxt/kit'
+import { normalize } from 'pathe'
+import MagicString from 'magic-string'
+
+interface LayerAliasingOptions {
+  sourcemap?: boolean
+  transform?: boolean
+  layers: NuxtConfigLayer[]
+}
+
+const ALIAS_RE = /(?<=['"])[~@]{1,2}(?=\/)/g
+
+export const LayerAliasingPlugin = createUnplugin((options: LayerAliasingOptions) => {
+  const aliases = Object.fromEntries(options.layers.map(l => [l.config.srcDir || l.cwd, {
+    '~': l.config.srcDir || l.cwd,
+    '@': l.config.srcDir || l.cwd,
+    '~~': l.config.rootDir || l.cwd,
+    '@@': l.config.rootDir || l.cwd,
+    ...l.config?.alias || {}
+  }]))
+  const layers = Object.keys(aliases)
+
+  return {
+    name: 'nuxt:layer-aliasing',
+    enforce: 'pre',
+    vite: {
+      resolveId: {
+        order: 'pre',
+        async handler (id, importer) {
+          if (!importer) { return }
+
+          const layer = layers.find(l => importer.startsWith(l))
```

Comment:
This could be refactored out to be shared between vite/webpack

Reply:
suggestions welcome, but I think the shared code is only this single line and refactoring would be more complex

Reply:
Logic can be bigger we don't want to match a node_modules path nested in a layer for example. I suggested this to make later works easier but we can do it later too :)